### PR TITLE
[cleanup][io] Cleanup Kafka connector

### PR DESCRIPTION
**Pull request by cbornet**
### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/cbornet/pulsar/pull/17

### CODE REVIEW

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java [hunk 1]
```diff
@@ -274,26 +274,27 @@ public Map<String, String> getProperties(){
             return properties;
         }
     }
-    protected static class KeyValueKafkaRecord<V> extends KafkaRecord implements KVRecord<Object, Object> {
 
-        private final Schema<Object> keySchema;
-        private final Schema<Object> valueSchema;
+    protected static class KeyValueKafkaRecord<K, W> extends KafkaRecord implements KVRecord<K, W> {
```
Comment (cbornet) [inline pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java:278]: We could do
```java
protected static class KeyValueKafkaRecord<K, W> extends KafkaRecord<KeyValue<K, W>> implements KVRecord<K, W>
```
If we could change KVRecord to
```java
public interface KVRecord<K, V> extends Record<KeyValue<K, V>>
```
But KVRecord is marked as `@InterfaceAudience.Public` `@InterfaceStability.Stable` so I don't think it's possible to change it ?

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java [hunk 2]
```diff
@@ -251,7 +251,7 @@ public Optional<Long> getRecordSequence() {
 
         @Override
         public Optional<String> getKey() {
-            return Optional.ofNullable(record.key());
+            return Optional.ofNullable(record.key() instanceof String ? (String) record.key() : null);
```
Comment (cbornet) [inline pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java:254]: Practically, the record key is always a String when `getKey()` is used.

**Review by nicoloboschi — APPROVED** [review]
LGTM

File: pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java (hunk 1 above)
Comment (eolivelli) [reply pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java:280]: It is not possible to change it, in the past I tried by it has some consequences I can't remember well

**Review by eolivelli — APPROVED** [review]

**Pull request merged**